cpacker/MemGPT
# feat: Add common + custom settings files for completion endpoints

Closes #623

---

## TODOs

- [x] Add two example common settings files
- [x] Make completions settings pull from common
- [x] Allow override from user directory (eg `~/.memgpt/settings/completions.py`)

Separate PR?
~~- [ ] Make openai/azure pull from common `/chat/completions` settings file~~
~~- [ ] Allow override from user directory (eg override temperature easily)~~

## Please describe the purpose of this pull request

- Right now if users want to modify the LLM inference settings they have to go into the package python files to open the constants
- All the completion-based backends should share a single base settings file
- This settings file should also be overwritten by settings that are placed somewhere in the `~/.memgpt` home directory

## How to test

Run inference via the backend and watch the backend endpoint to make sure settings are getting passed through properly (certain endpoints may require renaming of variables).

## Have you tested this PR?

- [ ] Test koboldcpp
- [ ] Test llamacpp
- [ ] Test webui
- [x] Test lmstudio
- [ ] Test vllm
